### DISCUSSION TITLE
Beginner question: How to get DHCP to support legacy and efi and avoid the endless loop

### DISCUSSION THREAD

**Discussion by jantypas** [ipxe/ipxe#1013] [Q&A]
I *hope* a simple question (from a simple person :-))....
Assume I have an isc-dhcp server, and it's serving both legacy and efi machines.
I assume I have to put something like
if ( prevent reboot loop) {
    if  (using ipxe) {
        if (using efi) {
           root-path = 
        } else {
           filename = 
        }
 }

Do we have example files?   What I'm trying to do is have a menu to offer the user:

- Boot Ubuntu server for install
- Boot FreeDOS (Bios flashes)
- Windows 10 install

**Comment by NiKiZe** [comment]
The steps described at https://ipxe.org/howto/dhcpd#pxe_chainloading should be correct.

There is also this nice config https://gist.github.com/robinsmidsrod/4008017
